agent/src/agent/pipeline/manager.py

```diff
@@ -23,17 +23,20 @@
 MAX_SAMPLE_RECORDS = 3
 
 
-class PipelineBuilder:
-    def __init__(self, pipeline_: Pipeline):
-        self.pipeline = pipeline_
-        self.file_loader = get_file_loader(pipeline_.source.type)()
-
-    def load_config(self, config, edit=False):
-        self.pipeline.set_config(self.file_loader.load(config, edit))
-        self._validate_config()
-
-    def _validate_config(self):
-        get_config_validator(self.pipeline).validate(self.pipeline)
+def get_default_protocol(pipeline_: Pipeline):
+    if pipeline_.protocol:
+        return pipeline_.protocol
+    else:
```

Comment:
you don't need to write `else` if you have a `return` at the end of `if`
Code with fewer nesting levels is easier to read

Reply:
yep